src/main/java/org/folio/consortia/service/UserTenantService.java

```diff
@@ -109,6 +111,24 @@ public interface UserTenantService {
 
   /**
    * Deletes orphaned shadow_users.
+   *
+   * @param userId id of user.
    */
   void deleteShadowUsers(UUID userId);
+
+  /**
+   * Get existing user.
+   *
+   * @param userId id of user.
+   */
+  User getUser(UUID userId);
+
+  /**
+   * Prepare shadow user from real user.
+   *
+   * @param userId id of user.
+   * @param tenantId id of tenant.
+   * @param folioExecutionContext folio context
+   */
+  User prepareShadowUser(UUID userId, String tenantId, FolioExecutionContext folioExecutionContext);
```

Comment:
Lets remain this method stateless, meaning it should not know in what folio execution context it invoked.
It is invoked now in 2 places, so you should move context setting to that places
`try (var context = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, folioExecutionContext))) {`

Reply:
fixed

Reply:
contracts now looks clear, thanks